support-lib/objc/DJFuture.mm

```diff
@@ -0,0 +1,170 @@
+/**
+  * Copyright 2021 Snap, Inc.
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License");
+  * you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *
+  *    http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+
+#import "DJFuture.h"
+
+@class DJSharedSate<Value>;
+
+typedef _Nullable id (^Continuation)(DJSharedSate* _Nonnull);
+
+@interface DJSharedSate<Value> : NSObject
+    @property (nonatomic, strong) Value value;
+    @property (nonatomic, strong) NSException* exception;
+    @property (nonatomic, strong) NSCondition* cond;
+    @property (nonatomic, strong) Continuation handler;
+
+    @property (readonly) BOOL isReady;
+@end
+
+@implementation DJSharedSate
+-(instancetype)init {
+    if (self = [super init]) {
+        self->_value = nil;
+        self->_exception = nil;
+        self->_cond = [[NSCondition alloc] init];
+        self->_handler = nil;
```

Comment:
```suggestion
        _cond = [[NSCondition alloc] init];
```

- `self->` not necessary
- all ivars are initialized to zeros by default. so, no need to initialized to `nil`.